pyracf/resource/resource_admin.py

```diff
@@ -285,6 +285,133 @@ def get_user_access(
         self.set_running_userid(original_userid)
         return self._get_field(profile, "base", "yourAccess")
 
+    # ============================================================================
+    # Auditing Rules
+    # ============================================================================
+    def get_audit_rules(
+        self, resource: str, class_name: str
+    ) -> Union[dict, bytes, None]:
+        """Get the auditing rules associated with this general resource profile."""
+        profile = self.extract(resource, class_name, profile_only=True)
+        return self._get_field(profile, "base", "auditing")
+
+    def overwrite_audit_by_attempt(
+        self,
+        resource: str,
+        class_name: str,
+        success: Union[str, None] = None,
+        failure: Union[str, None] = None,
+        all: Union[str, None] = None,
+    ) -> Union[dict, bytes]:
```

Comment:
I don't know if it makes sense to verify that the values the user gives here actually results in valid keys here? If someone specificies `success="doodlebob`, then we will end up with the trait `"base:audit_doodlebob" = "success`. The error will be caught downstream, but we it probably makes sense to check this early and say something like this:

```python
success = success.lower()
if success not in ("none", "read", "alter", "control"):
    # raise a new type of exception?
```

Reply:
For access level, I think we could probably let the downstream error handling handle it, but for attempt, I think it could make sense to do a check her since the value that the user gives ends up becoming a trait key. Downstream error handling should be executed to pick that up, but the error might be more confusing for the user in that case.

Reply:
That makes a lot of sense. I added functionality to raise a traditional `ValueError` and added unit testing that ensures this is raised for these functions.